### ISSUE TITLE
Is Microsoft Planning to fix ImageList?

### ISSUE THREAD

**Issue by SoftCircuits** [dotnet/msbuild#11185]
There is a lot of documentation about issue https://github.com/dotnet/winforms/issues/9701 and the security risks associated with `BinaryFormatter`.

But what if we're not using `BinaryFormatter` directly? Instead, we're using `ImageList`, which is currently implemented to use `BinaryFormatter`? Do we need to find a different image list control? Or is Microsoft planning to fix the existing one?

**Comment by Tanya-Solyanik** [comment]
@rainersigwald @Forgind  - here are the BF removal workgroup recommendations:

1. remove the MSB3825 warning when targeting .NET9+
2. reword this warning when targeting NET8:
     2.1 `may be used` -> `will be used`
     2.2 remove `of type {1}` - this type is not relevant to the author of the resx file, this is not the type that user serialized to the resx (in the above example, ImageListStreamer, the object inside the binary formatted payload, is not a string)

Resource "imageList1.ImageStream" **of type "System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" may be** deserialized via BinaryFormatter at runtime. BinaryFormatter is deprecated due to known security risks and is removed from .NET 9+. If you wish to continue using it, set property "GenerateResourceWarnOnBinaryFormatterUse" to false.
          More information: https://aka.ms/binaryformatter-migration-guide
-->
Resource "imageList1.ImageStream" **will be** deserialized via BinaryFormatter at runtime. BinaryFormatter is deprecated due to known security risks and is removed from .NET 9+. If you wish to continue using it, set property "GenerateResourceWarnOnBinaryFormatterUse" to false.
         More information: https://aka.ms/binaryformatter-migration-guide

**Comment by JanKrivanek** [comment]
@Tanya-Solyanik Just to clarify - when targeting lower than NET8 (or full FW) - then we do NOT want to warn - correct?

**Comment by Tanya-Solyanik** [comment]
@JanKrivanek 
>when targeting lower than NET8 (or full FW) - then we do NOT want to warn - correct?

Good point! Is this warning conditional on the target FX right now?

**Comment by JanKrivanek** [comment]
Yes - I believe it was to limit the update blockers - where just updating the toolchain would suddenly cause builds to fail. In such case we probably do not want to case disruptions to builds targeting NetFx (which we currently do not).

I'll then make sure to adjust wording for NET8 and remove the warning for NET9+

Thank you!
